enhancements/multi-arch/multiarch-manager-operator.md

```diff
@@ -0,0 +1,773 @@
+- user-provided `nodeSelectorTerms`: for each `nodeSelectorTerms`, all the `matchExpressions` are patched by adding the inferred `matchExpression` about the `kubernetes.io/arch`, except the ones already having it.
+
+#### Image inspection
+The inspection of images relies upon several machanisms, some of which depend on
+the specific scenario. The operator aims to execute in any Kubernetes
+distribution and will be user-config-driven, using the `PodPlacementConfig` CRD
+described in the previous sections. However, some behaviors can be automated to
+reduce the configuration effort for the user in Openshift specific scenarios.
+
+In particular, the inspection of an image may depend on:
+1. The credentials to access the registry where it's stored
+2. Any customized registry configurations[^6], i.e. ImageContentSourcePolicy,
+   ImageDigestMirrorSet in OCP
+3. The trusted CA bundle
+4. Other network or node customizations
+
+During phase 1, the operator will only deliver the support for 1 and 2, to
+access registries in connected scenarios. 2 and 3 will be delivered respectively
+during phase 2.  For the Openshift scenarios, we do not expect specific changes
+to be required to address 4, as most of those efforts are global across the
+installed clusters and managed by the machine config operator's resources.
+
+##### Registry authentication
+When pulling the images, the access to registries is done by merging (1) the
+global pull secret stored within a node, and (2) all the `dockerconfigjson`
+secrets listed by the pod spec's `ImagePullSecrets` list. The `ImagePullSecrets`
+fields are either provided by the users or filled automatically by the
+Kubernetes Pod Admission controllers through the ones associated with the
+service account (SA) running the pod.
+
+##### Registry reachability (proxy and mirroring)
+In phase 2, the `RegistriesMirrorCacher` will watch ICSPs/IDMSs to sync the
+registry mirror configurations within a map having a `source` as key and the
+list of `mirrors` as values.  The inspection of the images will change such that
+if the map has keys available, before trying to reach the original URI, the URI
+will be looked up in the map (the registry base URL only). If a hit occurs, the
+mirrors will be tried in order to inspect the image until the inspection
+succeeds or the list is fully visited. Finally, no changes are expected to be
+applied to support configurations like cluster-wide proxy at the time of
+writing.
+
+##### Registry and image verification
+While authenticating is critical to inspect images, so is the verification of
+the images and the registry being inspected. However, this is given lower
+priority as the operator is only inferring metadata from the image as opposed to
+pulling it.
+
+In Phase 1, the `skipTlsVerification` field is proposed in the CR and is set to
+true by default. It ignores verification of the TLS certificates the registry
```

Comment:
Exposing users’ credentials on the network is **completely unacceptable**.

Any version that leaves the original developers’ workstation simply must not default to disabling TLS. It would be irresponsible to include any code with that behavior into a shipped product.